js/main.js

```diff
@@ -0,0 +1,39 @@
+$(document).ready(function() {
+var total = 0;
+
+  function register() {
+    $("#entry").on("submit", function(pd){
+    pd.preventDefault();
+    var userInput = Number($("#newEntry").val());
+      if (userInput > 0) {
+        total += userInput;
+        var newEntry = $("<tr><td><td>"+userInput.toFixed(2)+"</td></td></tr>");
+        $("#entries").append(newEntry);
+        $("#total").html("$"+total.toFixed(2));
+        $("#entry").trigger("reset");
+      } else {
+        $("#entry").trigger("reset");
+      }
+    });
+  }
+
+  function dltButtons() {
+    $("#dlt").on("click", function(pd) {
+      pd.preventDefault();
+      $("#entries tr").each(function() {
+        $(this).children().first().html("<input type='submit' id='dlts' value='Delete Entry'>");
+      });
+    });
+  }
+
+  function dltEntry() {
+    $('#dlts').on('click',function(pd){
+        pd.preventDefault();
+        console.log("it works");
+      });
+  }
+
+register();
+dltButtons();
+dltEntry();
```

Comment:
this is being called after you call `dltButtons()`. I think the problem is that, the buttons for delete don't actually get created until after you click on the event on lines 21 - 26. So it is adding the event here, but no buttons actually exist until the other event gets fired.